src/model/NPMResolver.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2018, collaboration Factory AG. All rights reserved.
+ */
+
+import * as path from "path";
+import * as fs from "fs";
+import * as crypto from "crypto";
+import * as spawn from 'cross-spawn';
+import * as chokidar from "chokidar";
+import {Scheduler} from "../executor";
+import {cerr, cgreen, cred} from "../utils";
+import Timeout = NodeJS.Timeout;
+import * as rimraf from "rimraf";
+
+export class NPMResolver {
+    private static readonly PACKAGE_LOCK_HASH = 'package-lock.hash';
+    private static readonly PACKAGE_LOCK_JSON = 'package-lock.json';
+    private static readonly PACKAGE_JSON = 'package.json';
+    private static readonly NODE_MODULES = 'node_modules';
+    private mainRepo: string;
+    private readonly hashFilePath: string;
+
+    constructor(mainRepo: string, watch: boolean, clean: boolean) {
+        this.mainRepo = mainRepo;
+        if (clean) {
+            this.clean();
+        }
+        this.hashFilePath = this.getHashFilePath();
+        if (!this.isNPMInstallVersion()) {
+            console.log(cgreen`⇢` + ` [NPM] package.json version is 1.0.0 -> no npm install needed / your node modules are checked in`);
+        } else {
+            console.log(cgreen`⇢` + ` [NPM] package.json version is 2.0.0 or higher -> checking if npm install is needed`);
+            this.checkAndInstall();
+            if (watch) {
+                this.registerWatchers();
+            }
+        }
+    }
+
+    private isNPMInstallVersion(): boolean {
+        return !(this.getPackageVersion() === '1.0.0');
+    }
+
+    private clean() {
+        try {
+            let path: string = this.getNodeModulesPath();
+            rimraf.sync(path);
+            console.log(cgreen`⇢` + ` [NPM] node_modules folder was cleaned/removed`);
+        } catch (e) {
+            console.error(cerr` ${e.message}`);
+            throw Error(e);
+        }
+    }
+
+    private registerWatchers() {
+        // packageLockWatcher
+        const packageJsonWatcher = chokidar.watch([this.getPackageLockPath()]);
+        packageJsonWatcher
+            .on('change', () => {
+                this.checkAndInstall();
+            })
+            .on('error', (e) => {
+                console.error(cerr`✗` + ` [packageLockWatcher] watcher failed: ${e}`);
+                nodeModulesWatcher.close();
+            });
+
+        // nodeModulesWatcher
+        const glob = Scheduler.convertToUnixPath(`${this.getNodeModulesPath()}/**/*`);
```

Comment:
You should be able to just use `this.getNodeModulesPath()` and watch that single folder instead of all files inside of it - we just need to react to removing the `node_modules` folder itself.